### ISSUE TITLE
✨ [WIP] Add assessment step and proper sorting to Pool Candidates table

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#8960]
## ✨ Feature

In the Pool Candidates table...
Where the "Final Decision" column says "To Assess" it should also say which step needs to be assessed.
Sorting by Final Decision column or by Placement Status should work correctly, and not just use the old status sorting.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🎨 Design File

Include a link to the design file (if it exists).

### 🧑‍🎨 Designer

Include the author of the linked design file's name (or even better, their `@githubUsername`) for future reference.

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Criteria 1
- [ ] Criteria 2

## 🛑 Blockers
```[tasklist]
### Blocked By
- [ ] #8675 
- [ ] #8966 
```

**Comment by tristan-orourke** [comment]
Closing as a duplicate of #9792
